frontend/public/components/pdb/PDBForm.tsx

```diff
@@ -0,0 +1,381 @@
+  const converted = formValuesFromK8sResource(formData);
+  const [formValues, setFormValues] = React.useState(converted);
+  const [error, setError] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+  const [requirement, setRequirement] = React.useState('');
+  const [isDisabled, setDisabled] = React.useState(true);
+  const [labels, setLabels] = React.useState([]);
+  const [matchingSelector, setMatchingSelector] = React.useState<PodDisruptionBudgetKind>(null);
+  const [isOpen, setOpen] = React.useState(false);
+  const onToggle = (open: boolean) => setOpen(open);
+  const items = {
+    maxUnavailable: 'maxUnavailable',
+    minAvailable: 'minAvailable',
+  };
+
+  const existingData = React.useMemo(() => existingResource, [existingResource]);
+
+  const onFormValuesChange = React.useCallback(
+    (values) => {
+      setFormValues(values);
+      onChange(pdbToK8sResource(values));
+    },
+    [onChange],
+  );
+
+  React.useEffect(() => {
+    previousValueRef.current = SelectorInput.arrayify(selector);
+  });
+
+  React.useEffect(() => {
+    setRequirement(formValues.requirement);
+    if (!_.isEmpty(existingData) && _.isEmpty(formValues.name)) {
+      setFormValues(formValuesFromK8sResource(existingData));
+    }
+    if (!_.isEqual(SelectorInput.arrayify(selector), previousValue)) {
+      setLabels(SelectorInput.arrayify(selector));
+      onFormValuesChange({ ...formValues, selector: { matchLabels: selector } });
+    }
+  }, [existingData, formValues, onFormValuesChange, previousValue, selector]);
+
+  const handleNameChange = (value: string) => onFormValuesChange({ ...formValues, name: value });
+
+  const handleSelectorChange = (labelValues: string[]) => {
+    const filterMatchLabels = labelValues.filter((f) => f.includes('='));
+    const filterMatchExpressions = labelValues.filter((f) => !f.includes('='));
+    onFormValuesChange({
+      ...formValues,
+      selector: {
+        matchLabels: SelectorInput.objectify(filterMatchLabels),
+        matchExpressions: SelectorInput.arrayToArrayOfObjects(filterMatchExpressions),
+      },
+    });
+    setLabels([...labels, ...labelValues]);
+    setMatchingSelector(existingData);
+  };
+
+  const handleAvailabilityRequirementKeyChange = (value: string) => {
+    setRequirement(value);
+    setDisabled(false);
+    setOpen(!isOpen);
+    onFormValuesChange({
+      ...formValues,
+      requirement: value,
+      minAvailable: '',
+      maxUnavailable: '',
+    });
+  };
+  const handleAvailabilityRequirementValueChange = (value: string | number) => {
+    onFormValuesChange({ ...formValues, [requirement]: value });
+  };
+
+  const disableSubmit =
+    formValues.name === '' ||
+    _.isEmpty(formValues.selector) ||
+    (formValues.minAvailable === '' && formValues.maxUnavailable === '');
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setInProgress(true);
+
+    const patch: Patch[] = [];
```

Comment:
nit: It would be cleaner to pull the patch logic into a separate method and only call if there is an existing resource. You could have `patchPDB` and `createPDB` methods called from `handleSubmit`.